de.dlr.sc.virsat.model/src/de/dlr/sc/virsat/model/dvlm/structural/util/StructuralElementInstanceHelper.java

```diff
@@ -44,13 +44,12 @@ public StructuralElementInstanceHelper(StructuralElementInstance sei) {
 	 * 
 	 * @return the structural element instance as the root
 	 */
-	public  ICategoryAssignmentContainer getRoot() {
-
-		StructuralElementInstance eContainer = sei.getParent();
-		while ((eContainer != null) && (eContainer.getParent() != null) && (eContainer.getParent() instanceof StructuralElementInstance)) {
-			eContainer = eContainer.getParent();
+	public ICategoryAssignmentContainer getRoot() {
+		StructuralElementInstance root = sei;
+		while (root.getParent() != null) {
+			root = root.getParent();
 		}	
-		return (ICategoryAssignmentContainer) ((eContainer ==  null) ? sei : eContainer);
+		return root;

```

Comment:
Is it really that simple now? Why did we have so much more complicated before?

Reply:
It is. As for why it was so complicated before: I guess maybe in the past getParent of a StructuralElementInstance didnt give a SEI but an ICategoryAssignmentContainer. In that case, the old implementation makes sense because some type checking is needed. But since getParent gives a SEI, removing all the type checking logic gives the simple implementation